tests/php/sync/test_class.jetpack-sync-settings.php

```diff
@@ -47,4 +47,42 @@ function test_settings_disable_enqueue_and_clears_queue() {
 
 		Jetpack_Sync_Settings::update_settings( array( 'disable' => 0 ) );
 	}
+
+	function test_settings_disable_network_enqueue_and_clears_queue() {
+		if ( ! is_multisite() ) {
+			$this->markTestSkipped( 'Not compatible with single site mode' );
+		}
+
+		$event = $this->server_event_storage->reset();
+
+		// create a post - this will end up in the queue before data is sent
+		$post_id = $this->factory->post->create();
+		$this->assertTrue( $this->listener->get_sync_queue()->size() > 0 );
+
+		Jetpack_Sync_Settings::update_settings( array( 'network_disable' => 1 ) );
+
+		// generating posts should no longer affect queue size
+		$this->assertEquals( 0, $this->listener->get_sync_queue()->size() );
```

Comment:
In case we haven't cleaned up properly from previous tests (which seems to be the case here), it might make more sense to omit this and just retrieve `$this->listener->get_sync_queue()->size()` in a parameter; then in the next assertion compare the size with the previous one. What do you think?

Reply:
I am bit torn here. Do we rest the queue when we disable the sync network setting? 
To do it properly we would have to clear it on all the sites 🤔

Reply:
Yeah, I agree it would be best to clear it every time, but there might be tests we might break that way.